pkg/comp-functions/functions/common/non_sla_prom_rules.go

```diff
@@ -135,6 +102,44 @@ func generatePromeRules(elem InfoGetter, svc *runtime.ServiceRuntime) error {
 		},
 	}
 
+	if serviceName != "keycloak" {
```

Comment:
Only MemoryCritical is active for keycloak? Also I don't like this `if` in common. It's not suppose to work like that. Either find a way to pass this from `keycloak` package or don't use _common_ package.

Reply:
The problem is, Keycloak is really the only service we have that stores completely nothing, it just connects to DB. Alternative solution would be to copy-paste file for vshnkeycloak and adjust is for just keycloak - which I find even worse than `if` statement. This way it's consistent, file in common solves issue for all of the services, if we introduce some day another service that doesn't use any storage, we then append `if` statement

Reply:
{quote}
Alternative solution would be to copy-paste file for vshnkeycloak and adjust is for just keycloak
{quote}
Why would you copy paste files? Just change the code in such a way to not copy paste and not use if clause. Change or add functions definition that would achieve that.

Reply:
I would go with this solution `Another one could be to define a map of service - list of rules in a separate file.` You create a new struct like `ServicePromRule` which has a map inside. In that map you add the rules per each service. Since most services use the same set of rules you don't copy paste anything. Then you can pass this object `ServicePromRule` to  `GenerateNonSLAPromRules()` or figure another way how to do it.

Reply:
I like the map idea.

Reply:
`
type ServicePromRule struct {
	a map[string][]promV1.Rule
}
`

Then you initialise and populate it somewhere in a new file named `service-prom-rules.go`. That way we have all the rules per each service clearly displayed. 

OR

`
type ServicePromRule struct {
	service string
	r []promV1.Rule
}
`
And you add it to the definition of the `GenerateNonSLAPromRules()`

Reply:
I was just thinking about this for a bit longer. This kind of refactoring will help us get out from the business logic only the code changes based on the service we deploy. At some point we will have enough "generic" code that we might be able to create a framework that in the end will ask in input the set of prometheus rules to deploy for the service.